# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/mine/rocket.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Index: mine/rocket.sh
===================================================================
--- mine/rocket.sh	(revision 58)
+++ mine/rocket.sh	(working copy)
@@ -43,7 +43,7 @@
 chunk == 2 { print > "/var/adm/rocket/dependencies/" pkg; }
 chunk == 3 { print > "/var/adm/rocket/cksums/" pkg; }
 
-chunk == 1 && $1 == "[V]" { ver=$2; }
+chunk == 1 && $1 == "[V]" { ver = $2 "-" $3; }
 
 ' < <( curl -s "$1/packages.db" | gunzip; )
 }
@@ -69,7 +69,8 @@
     create_gem)
 	shift
 	for pkg; do
-		v="$(grep '^Package Name and Version' /var/adm/packages/$pkg | cut -f6 -d' ')"
+		v="$(grep '^Package Name and Version' /var/adm/packages/$pkg | \
+				cut -f6,7 -d' ' | tr ' ' - )"
 		mine -T /var/adm / "$pkg" "$pkg-$v.tar.bz2"
 		mine -C /var/adm "$pkg-$v.tar.bz2" "$pkg" "$pkg-$v.gem"
 	done
@@ -93,7 +94,8 @@
     install)
 	shift
 	for x; do
-		url="$( grep "^$x " /var/adm/rocket/locations.txt | cut -f2 -d' ' | tail -1; )"
+		url="$( grep "^$x " /var/adm/rocket/locations.txt | \
+				cut -f2 -d' ' | tail -n1; )"
 		if [ -z "$url" ]; then
 			echo "Package $x not found - maybe you need to add the archive"
 			echo "to /etc/rocket.conf and run 'rocket update' first."
